src/pages/settings/Profile/CustomStatus/StatusClearAfterPage.tsx

```diff
@@ -1,59 +1,49 @@
-import _ from 'lodash';
-import lodashGet from 'lodash/get';
-import PropTypes from 'prop-types';
 import React, {useCallback, useEffect, useMemo, useState} from 'react';
 import {View} from 'react-native';
 import {withOnyx} from 'react-native-onyx';
+import type {OnyxEntry} from 'react-native-onyx';
 import FormProvider from '@components/Form/FormProvider';
 import HeaderWithBackButton from '@components/HeaderWithBackButton';
 import MenuItemWithTopDescription from '@components/MenuItemWithTopDescription';
 import ScreenWrapper from '@components/ScreenWrapper';
 import RadioListItem from '@components/SelectionList/RadioListItem';
 import Text from '@components/Text';
-import withCurrentUserPersonalDetails, {withCurrentUserPersonalDetailsDefaultProps} from '@components/withCurrentUserPersonalDetails';
-import withLocalize from '@components/withLocalize';
+import useCurrentUserPersonalDetails from '@hooks/useCurrentUserPersonalDetails';
 import useLocalize from '@hooks/useLocalize';
 import useThemeStyles from '@hooks/useThemeStyles';
 import * as User from '@libs/actions/User';
-import compose from '@libs/compose';
 import DateUtils from '@libs/DateUtils';
 import Navigation from '@libs/Navigation/Navigation';
 import * as ValidationUtils from '@libs/ValidationUtils';
-import personalDetailsPropType from '@pages/personalDetailsPropType';
 import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
 import ROUTES from '@src/ROUTES';
+import type * as OnyxTypes from '@src/types/onyx';
 
-const defaultProps = {
-    ...withCurrentUserPersonalDetailsDefaultProps,
-};
+type CustomStatusTypes = (typeof CONST.CUSTOM_STATUS_TYPES)[keyof typeof CONST.CUSTOM_STATUS_TYPES];
 
-const propTypes = {
-    currentUserPersonalDetails: personalDetailsPropType,
-    customStatus: PropTypes.shape({
-        clearAfter: PropTypes.string,
-    }),
+type StatusClearAfterPageOnyxProps = {
+    customStatus: OnyxEntry<OnyxTypes.CustomStatusDraft>;
 };
 
+type StatusClearAfterPageProps = StatusClearAfterPageOnyxProps;
+
 /**
- * @param {string} data -  either a value from CONST.CUSTOM_STATUS_TYPES or a dateTime string in the format YYYY-MM-DD HH:mm
- * @returns {string}
+ * @param data -  either a value from CONST.CUSTOM_STATUS_TYPES or a dateTime string in the format YYYY-MM-DD HH:mm
  */
-function getSelectedStatusType(data) {
+function getSelectedStatusType(data: string): CustomStatusTypes {
     switch (data) {
         case DateUtils.getEndOfToday():
             return CONST.CUSTOM_STATUS_TYPES.AFTER_TODAY;
         case CONST.CUSTOM_STATUS_TYPES.NEVER:
         case '':
             return CONST.CUSTOM_STATUS_TYPES.NEVER;
-        case false:
```

Comment:
Are you sure this isn't used?

Reply:
The only place where `getSelectedStatusType` is used is in this file in `useState` where two strings are passed, so I'm quite sure

Reply:
Let's test manually and see if `data` argument is ever a boolean

Reply:
Checked manually and it seems that it is always a string